app/models/pieces/piece.rb

```diff
@@ -15,15 +15,28 @@ def obstructed?(x_destination, y_destination)
     pieces = []
     x = x_position
     y = y_position
-
     if moving_horizontally?(y_destination)
-      x = (x_position + 1..x_destination - 1)
+      return false unless diff(x_destination, x_position) > 1
+      direction_start_x = x_destination <=> x_position
+      x_start = x_position + direction_start_x
+      x_end = x_destination - (diff(x_destination, x_position) > 1 ? direction_start_x : 0)
+      x = x_start < x_end ? x_start..x_end : x_end..x_start
```

Comment:
@lasley x and y are used in the Piece.where statements on line 43. I used the x_start, x_end, etc variables to split out the logic over multiple lines because I thought it would make the code more readable. Would it be better to refactor this code to use less variables? I am not sure how to begin with that.